### ISSUE TITLE
Thumbnails not created when uploading a new dataset on Nexus dev server

### ISSUE THREAD

**Issue by giohappy** [geosolutions-it/nexus-geonode#389]
Please @mattiagiupponi take a look at what's happening non Nexus dev server on this regard,
You can get the credentials [here](https://docs.google.com/document/d/1P0M94t6QWqvngp0KqTY44KshQbRIUwMjbqb5OwTtiS4/).
You can ask @randomorder to add your key for SSH access.

**Comment by randomorder** [comment]
Please hand me your public SSH keys @mattiagiupponi

**Comment by randomorder** [comment]
access granted

**Comment by mattiagiupponi** [comment]
After some analysis, looks like that the Thumbnail is generated, but in some way is not assigned to the dataset.
This is an example:
This dataset is without thumbnail
https://dev.nexus.geosolutionsgroup.com/catalogue/#/detail/dataset/101

but checking the logs of celery, technically the thumbnail is created (downloaded from the thumb/ folder)

![dataset-9c9a707e-f119-11eb-95d4-0242ac180006-thumb_IGDLuLw](https://user-images.githubusercontent.com/51856725/127637620-7cc641e6-cfb5-42ec-b6bd-38bd32749fb4.png)

I'm keeping investigating.

**Comment by mattiagiupponi** [comment]
UPDATE: After some deeper analysis, it looks like that there is an issue regarding the thumbnail file name during the creation of it.

Celery logs those errors:

```
[2021-07-30 11:11:36,127: ERROR/ForkPoolWorker-1104] Error when generating the thumbnail for resource 105. (Generated thumbnail image is zero size)
[2021-07-30 11:11:36,127: ERROR/ForkPoolWorker-1104] Check permissions for file thumbs/dataset-dfb7bcf6-f126-11eb-aeaa-0242ac180006-thumb.png.
```
The function tries to check the path for a thumbnail named: `dataset-dfb7bcf6-f126-11eb-aeaa-0242ac180006-thumb.png`, while the file created and saved into the storage has this name: `dataset-9c9a707e-f119-11eb-95d4-0242ac180006-thumb_IGDLuLw.png`
There is a misalignment between the name extracted and the one returned by the storage manager.

**Comment by mattiagiupponi** [comment]
Fix ready for review here: https://github.com/GeoNode/geonode/pull/7897

**Comment by mattiagiupponi** [comment]
> Fix ready for review here: [GeoNode/geonode#7897](https://github.com/GeoNode/geonode/pull/7897)

Looks like the issue is still present, still investigating

**Comment by marthamareal** [comment]
I was able to reproduce this locally on running the project with Docker, When i check the logs, i see get a connection error when they try to call wms.get_map.

**Comment by marthamareal** [comment]
> UPDATE: After some deeper analysis, it looks like that there is an issue regarding the thumbnail file name during the creation of it.
> 
> Celery logs those errors:
> 
> ```
> [2021-07-30 11:11:36,127: ERROR/ForkPoolWorker-1104] Error when generating the thumbnail for resource 105. (Generated thumbnail image is zero size)
> [2021-07-30 11:11:36,127: ERROR/ForkPoolWorker-1104] Check permissions for file thumbs/dataset-dfb7bcf6-f126-11eb-aeaa-0242ac180006-thumb.png.
> ```
> 
> The function tries to check the path for a thumbnail named: `dataset-dfb7bcf6-f126-11eb-aeaa-0242ac180006-thumb.png`, while the file created and saved into the storage has this name: `dataset-9c9a707e-f119-11eb-95d4-0242ac180006-thumb_IGDLuLw.png`
> There is a misalignment between the name extracted and the one returned by the storage manager.

On addition to @mattiagiupponi  comment, It seems in docker environment. files with the created thumbnail_name are not saved but only temp files are saved in the MEDIA_ROOT. Am trying to debug with docker but am still getting error in starting the debug container.

**Comment by mattiagiupponi** [comment]
Looks like that the issue is a matter of double slashes in the `save_thumbnail` function.
Here is the PR on GeoNode https://github.com/GeoNode/geonode/pull/7975, I'm testing it locally.
cc: @marthamareal

**Comment by marthamareal** [comment]
@giohappy this PR seems to solve this issue locally https://github.com/GeoNode/geonode/pull/7975, was it deployed on the Nexus dev server so that i can if the issue is resolved.

**Comment by giohappy** [comment]
I'm not sure if the last build includes the fix @marthamareal. I'm going to rebuild the images. I will notify you when ready.

**Comment by giohappy** [comment]
@marthamareal dev necus updated, please make your checks.

**Comment by marthamareal** [comment]
@giohappy  Yes thumbnails are now created. [example](https://dev.nexus.geosolutionsgroup.com/catalogue/#/detail/dataset/127?filter%7Bresource_type.in%7D=dataset)

**Comment by giohappy** [comment]
well done @marthamareal. Can you check if we have a command to automatically recreate thumbnails for existing datasets?
